test/src/vfs_test_helper.h

```diff
@@ -0,0 +1,443 @@
+/**
+ * @file   vfs_test_helper.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2020 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file declares some vfs-specfic test suite helper functions.
+ */
+
+#ifndef TILEDB_VFS_TEST_HELPER_H
+#define TILEDB_VFS_TEST_HELPER_H
+
+#include "test/src/helpers.h"
+
+#ifdef _WIN32
+#include "tiledb/sm/filesystem/win.h"
+#else
+#include "tiledb/sm/filesystem/posix.h"
+#endif
+
+using namespace tiledb::common;
+
+namespace tiledb {
+
+namespace sm {}
+
+namespace test {
+
+/**
+ * This class defines and manipulates
+ * a list of supported filesystems.
+ */
+class SupportedFs {
+ public:
+  SupportedFs()
+      : temp_dir_()
```

Comment:
We only _need_ to initialize primitives in the constructor. `temp_dir_` is a string, so we do not need to default-initialize it here: the default constructor will be implicitly used.